plugins/providers/bigquery/provider.go

```diff
@@ -175,7 +176,11 @@ func (p *Provider) GetResources(pc *domain.ProviderConfig) ([]*domain.Resource,
 					table.ProviderURN = pc.URN
 					children[i] = table
 				}
-				dataset.Children = children
+				if containsString(resourceTypes, ResourceTypeDataset) {
+					dataset.Children = children
+				} else {
+					resources = append(resources, children...)
+				}
```

Comment:
There's no table filters in this logic. i.e, if one adds a table resource filter in provider it will not work. 
Does this mean we are not addressing the fix of table filters as part of this MR  ?

Reply:
As per our discussion -
* The filter happens at provider level also in [here](https://github.com/goto/guardian/blob/main/core/provider/service.go#L552), so this is not a problem from that angle.
* But there is a performance bottleneck that would arise similar to the one that needed changes in this PR - https://github.com/goto/guardian/pull/82
   - Issue details: If the user has set a table filter like `resource.urn contains dataset_name`. The code will get all tables in a given bigquery project and then at provider level, it will apply filter. If a project has many tables, this will take a lot of time and be a performance problem.

@abhishekv24 @rahmatrhd

Reply:
Issue: https://github.com/goto/guardian/issues/113